usaspending_api/search/delta_models/award_search.py

```diff
@@ -291,8 +291,32 @@
   COALESCE(transaction_fpds.legal_entity_county_name, transaction_fabs.legal_entity_county_name) AS recipient_location_county_name,
   LPAD(CAST(CAST(REGEXP_EXTRACT(COALESCE(transaction_fpds.legal_entity_congressional, transaction_fabs.legal_entity_congressional), '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 2, '0')
             AS recipient_location_congressional_code,
-  -- TODO: include recipient_location_congressional_code_current derivation
-  'TEST CUR REC CONGR AS' AS recipient_location_congressional_code_current,
+  (CASE
+      WHEN (
+          COALESCE(transaction_fpds.legal_entity_country_code, transaction_fabs.legal_entity_country_code) <> 'USA'
+      ) THEN NULL
+      WHEN (
+          COALESCE(transaction_fpds.legal_entity_country_code, transaction_fabs.legal_entity_country_code) = 'USA'
+          AND rl_state_cd.congressional_district_no IS NOT NULL
+      ) THEN rl_state_cd.congressional_district_no
+      WHEN (
```

Comment:
There's a lot of code duplication here that I think you can reduce.

You could reduce the number of times this predicate, COALESCE(transaction_fpds.legal_entity_country_code, transaction_fabs.legal_entity_country_code) = 'USA', is duplicated by using a COALESCE.

For example, to lines 302 to 317 can be reduced to only 4 lines.

`
```
WHEN (
          COALESCE(transaction_fpds.legal_entity_country_code, transaction_fabs.legal_entity_country_code) = 'USA'
          AND COALESCE(rl_state_cd.congressional_district_no, rl_zips.congressional_district_no, rl_cd_zips_grouped.congressional_district_no, l_cd_city_grouped.congressional_district_no, rl_cd_county_grouped.congressional_district_no) IS NOT NULL
      ) THEN COALESCE(rl_state_cd.congressional_district_no, rl_zips.congressional_district_no, rl_cd_zips_grouped.congressional_district_no, l_cd_city_grouped.congressional_district_no, rl_cd_county_grouped.congressional_district_no)
```

Reply:
Good catch! I had to test it out to make sure everything worked and glad it worked out. Updated accordingly.